php-frontend/src/main/java/org/sonar/php/parser/NewPHPGrammar.java

```diff
@@ -0,0 +1,1222 @@
+            LABEL()
+        ));
+  }
+
+  public StaticStatementTree STATIC_STATEMENT() {
+    return b.<StaticStatementTree>nonterminal(PHPLexicalGrammar.STATIC_STATEMENT)
+        .is(f.staticStatement(
+            b.token(STATIC),
+            STATIC_VAR(),
+            b.zeroOrMore(f.newTuple22(b.token(COMMA), STATIC_VAR())),
+            EOS()
+        ));
+  }
+
+  public VariableDeclarationTree STATIC_VAR() {
+    return b.<VariableDeclarationTree>nonterminal(PHPLexicalGrammar.STATIC_VAR)
+        .is(f.staticVar(
+            b.token(PHPLexicalGrammar.REGULAR_VAR_IDENTIFIER),
+            b.optional(f.newTuple24(b.token(EQU), STATIC_SCALAR()))
+        ));
+  }
+
+  public DeclareStatementTree DECLARE_STATEMENT() {
+    return b.<DeclareStatementTree>nonterminal(PHPLexicalGrammar.DECLARE_STATEMENT)
+        .is(b.firstOf(
+            f.shortDeclareStatement(
+                DECLARE_STATEMENT_HEAD(),
+                EOS()
+            ),
+            f.declareStatementWithOneStatement(
+                DECLARE_STATEMENT_HEAD(),
+                STATEMENT()
+            ),
+            f.alternativeDeclareStatement(
+                DECLARE_STATEMENT_HEAD(),
+                b.token(COLON),
+                b.zeroOrMore(INNER_STATEMENT()),
+                b.token(PHPKeyword.ENDDECLARE),
+                EOS()
+            )
+        ));
+  }
+
+  public DeclareStatementHead DECLARE_STATEMENT_HEAD() {
+    return b.<DeclareStatementHead>nonterminal()
+        .is(f.declareStatementHead(
+            b.token(PHPKeyword.DECLARE),
+            b.token(LPARENTHESIS),
+            MEMBER_CONST_DECLARATION(),
+            b.zeroOrMore(f.newTuple20(b.token(COMMA), MEMBER_CONST_DECLARATION())),
+            b.token(RPARENTHESIS)
+        ));
+  }
+
+  public InlineHTMLTree INLINE_HTML() {
+    return b.<InlineHTMLTree>nonterminal(PHPLexicalGrammar.INLINE_HTML_STATEMENT)
+        .is(f.inlineHTML(b.token(PHPLexicalGrammar.INLINE_HTML)));
+  }
+
+  public StatementTree INNER_STATEMENT() {
+    return b.<StatementTree>nonterminal(PHPLexicalGrammar.INNER_STATEMENT)
+        .is(b.firstOf(
+            FUNCTION_DECLARATION(),
+            CLASS_DECLARATION(),
+            TRAIT_DECLARATION(),
+            INTERFACE_DECLARATION(),
+            STATEMENT()
+        ));
+  }
+
+
+  public GlobalStatementTree GLOBAL_STATEMENT() {
+    return b.<GlobalStatementTree>nonterminal(PHPLexicalGrammar.GLOBAL_STATEMENT)
+        .is(f.globalStatement(
+            b.token(PHPKeyword.GLOBAL),
+            GLOBAL_VAR(),
+            b.zeroOrMore(f.newTuple16(b.token(COMMA), GLOBAL_VAR())),
+            EOS()
+        ));
+  }
+
+  public VariableTree GLOBAL_VAR() {
+    return b.<VariableTree>nonterminal(PHPLexicalGrammar.GLOBAL_VAR).is(
+      f.globalVar(
+        b.zeroOrMore(b.token(PHPLexicalGrammar.VARIABLE_VARIABLE_DOLLAR)),
+        COMPOUND_VARIABLE()));
+  }
+
+  public UnsetVariableStatementTree UNSET_VARIABLE_STATEMENT() {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Rename this method name to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS00100)